nilearn/input_data/tests/test_niftispheresmasker.py

```diff
@@ -0,0 +1,38 @@
+import numpy as np
+import nibabel
+
+from nilearn.input_data import NiftiSpheresMasker
+#Preparing img with nans inside it 
+def generate_imgs():
+    affine = np.eye(4)
+
+    data_with_nans = np.random.random((10, 10, 10))
```

Comment:
if you fill the values with nans, you shold not create it with random call: use e.g. zeros instead